src/rpc/connectivity/cluster.cc

```diff
@@ -409,8 +422,10 @@ void connectivity_cluster_t::run_t::join_blocking(
                    &peer,
```

Comment:
Actually I think we need to find a way to report permanent failures on a per-IP basis. Otherwise a single wrong IP for a server (e.g. if `127.0.0.1` ends up in the IP address list) can lead to us stopping to reconnect to the server altogether.

In the `auto_reconnector_t`, we must then only remove the individual IPs that failed from the addresses set in the `peer_address_t` structure. Only when that set becomes empty should we stop reconnecting to the server altogether I think.